### PR TITLE
PLT-5100 add getUsersStats

### PR DESCRIPTION
**Pull request by jsfeb26**
Please make sure you've read the [pull request](http://docs.mattermost.com/developer/contribution-guide.html#preparing-a-pull-request) section of our [code contribution guidelines](http://docs.mattermost.com/developer/contribution-guide.html).

When filling in a section please remove the help text and the above text.

#### Summary
Adds `getUsersStats` needed for mattermost/platform#6614

#### Ticket Link
JIRA Ticket: https://mattermost.atlassian.net/browse/PLT-5100
Github Issue: #6521 

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] Added or updated unit tests (required for all new features)

#### Test Information
This PR was tested on: [Device name(s), OS version(s)]

### CODE REVIEW

**Comment by enahum** [comment]
thanks @jsfeb26 

Please do **NOT** merge before https://github.com/mattermost/mattermost-redux/pull/189

File: src/reducers/entities/users.js [hunk 1]
```diff
@@ -293,6 +293,22 @@ function statuses(state = {}, action) {
     }
 }
 
+function stats(state = {}, action) {
+    switch (action.type) {
+    case UserTypes.RECEIVED_USER_STATS: {
+        const stat = action.data;
+        return {
+            ...state,
+            ...stat
```
Comment (enahum) [inline src/reducers/entities/users.js]: What would be the keys of this object? sorry I just can't see it

File: src/selectors/entities/users.js [hunk 2]
```diff
@@ -65,23 +69,6 @@ export function getUserByUsername(state, username) {
     return getUsersByUsername(state)[username];
 }
 
-export const getUsersByEmail = createSelector(
```
Comment (enahum) [inline src/selectors/entities/users.js]: Why are these selectors being removed?

File: test/actions/users.test.js [hunk 3]
```diff
@@ -351,6 +351,26 @@ describe('Actions.Users', () => {
...
+        nock.restore();
+
+        const {stats} = store.getState().entities.users;
+        const statsRequest = store.getState().requests.users.getUsersStats;
+
+        if (statsRequest.status === RequestStatus.FAILURE) {
+            throw new Error(JSON.stringify(statsRequest.error));
+        }
+
+        assert.equal(stats.total_user_count, 656);
```
Comment (enahum) [inline test/actions/users.test.js]: this test seems to be very error prone, what happens if we add more tests, this number surely will fail

**Comment by esethna** [comment]
@jsfeb26 just checking in on this one, anything we can help clarify?

**Comment by jsfeb26** [comment]
@enahum sorry, I haven't gotten to take a look yet. I'll try to address your comments sometime this week.

**Comment by enahum** [comment]
@jsfeb26 this PR will also need a rebase

**Comment by esethna** [comment]
@jsfeb26 let us know if there is anything we can help clarify so we can get this merged

**Comment by esethna** [comment]
Closing this PR due to inactivity, please let us know if you're interested in continuing to work on this project and we can re-open.

**Pull request closed without merging**